lib/files.h

```diff
@@ -628,4 +632,16 @@ bool files_load_attest_file(FILE *f, const char *path, TPMS_ATTEST *attest);
 tool_rc files_load_unique_data(const char *file_path,
 TPM2B_PUBLIC *public_data);
 
+/**
+ * @brief
+ * Check whether file can be read
```

Comment:
this is introduces a Time of Check Time of Use Bug. The best way to detect if a file cannot be read is when fopen() fails when you go to read it. Why do we need this?

Reply:
The idea for the check was that the flush should only be executed when a context file for the parent exists. If only the TPM handle for the parent was passed the flash should not be executed.
I wanted to avoid calling the close function after the file test. Therefore i added this function.

Reply:
OK so the problem is, a user passes in a transient handle not in a context file and under those conditions we don't flush it. I guess this is possible when an RM is not being used, did I grok this right?

I think the right fix is to look at the `tpm2_loaded_object` structure s path field, it should be NULL if I understand the flow in `tpm2_util_object_load2`, as it calls `tpm2_util_object_load2` only if the `fopen` succeeds and that function sets the objects path. So if that path is set, then it's a context file. You could also add an attribute to the object `bool is_ctx_file` and set that and then check that over checking the path. Might be more robust as code changes.

Reply:
* yes you did grok it right. 
* I have added "1" to the possible values of the env variable.
* It's a good idea to check whether the parent path is a context file. Because there is no common structure for the context of the affected commands I would prefer to add the check to the file_exists function. To make the use of the function clearer, I have renamed the function to `ctx_file_exists`.